docs/proposals/promote-dependencies-automatically/README.md

```diff
@@ -0,0 +1,196 @@
+2. Determine whether to enable the functionality of automatically promoting all dependent resources based on the value of `Deps`. 
+   1. If `Deps` is set to `false`, only create the corresponding Resource Template, ClusterPropagationPolicy/PropagationPolicy, and generate the respective ResourceBinding in the Karmada control plane for the workload.
+   2. If `Deps` is set to `true`, use the interpret framework to analyze the dependencies of the corresponding workload. Create the Resource Templates, ClusterPropagationPolicy/PropagationPolicy, and ResourceBindings for the dependencies in the Karmada control plane. Then, create the ResourceBinding for the workload itself. 
+
+**Note:**
+If the workload to be promoted is a CR(Custom Resource), we need to first promote its CRD (Custom Resource Definition).
+
+### Interpret Workload Dependencies
+
+To achieve the automatic promotion of dependent resources, the key challenge is to determine how to obtain the dependent resources of the resource to be promoted. Therefore, we need to be able to parse the dependencies of a resource. Currently, the interpret framework in Karmada supports the following methods for parsing dependent resources:
+
+1. Default built-in interpretation
+   1. Native Kubernetes Resource Interpretation:
+      1. Perform dependency parsing using the structure of native Kubernetes resources.
+   2. Third-Party Resource Interpretation:
+      1. Pre-configure Lua scripts for commonly used CR(Custom Resource) to enable dependency parsing using Lua configurations.
+3. Custom Interpretation
+   1. Lua Configuration-based Interpretation:
+      1. Configure resource interpretation rules for dependency parsing by specifying Lua scripts in the [ResourceInterpreterCustomization](https://github.com/karmada-io/karmada/blob/84b971a501ba82c53a5ad455c2fe84d842cd7d4e/pkg/apis/config/v1alpha1/resourceinterpretercustomization_types.go#L17) resource.
+   2. Webhook-based Interpretation:
+      1. Users can design Resource Interpreter Webhooks for specific custom resources to handle Interpreter Operations requests and perform dependency parsing. The process involves receiving these request operations and processing them accordingly.
+         The specific process is as follows:
+
+![webhook-interpret](statics/webhook-interpret.png)
+
+***Note:*** 
+Using the webhook-based interpretation method may pose difficulties when making requests to the webhook interpret service. In such cases, we need to perform port forwarding to invoke the service within the cluster.
+
+### Conflict and Failure Resolution
+
+#### Conflict Resolution
+
+When promoting a resource, if the resource to be promoted or its dependent resources have already been promoted to the Karmada control plane, the promotion operation will abort the promotion of the resource or its dependent resources. The user needs to delete the existing resource in the Karmada control plane before promoting it again.
```

Comment:
I think we should skip the resource that already exists, instead of abortion.

What do you think? @chaunceyjiang @XiShanYongYe-Chang

Reply:
+1

Reply:
Perhaps interrupting or skipping could be left to the user's choice.

Reply:
> I think we should skip the resource that already exists, instead of aborting.

I tend to abort because in case resources already exist, we can't ensure the content of the resource(like Secret/ConfigMap) is exactly the same as in the member cluster.